qiskit_aer/primitives/estimator_v2.py

```diff
@@ -0,0 +1,146 @@
+from qiskit.primitives.containers.estimator_pub import EstimatorPub
+from qiskit.primitives.primitive_job import PrimitiveJob
+from qiskit.quantum_info import Pauli
+
+from qiskit_aer import AerSimulator
+
+
+@dataclass
+class Options:
+    """Options for :class:`~.EstimatorV2`."""
+
+    default_precision: float = 0.0
+    """The default precision to use if none are specified in :meth:`~run`.
+    """
+
+    backend_options: dict = field(default_factory=dict)
+    """backend_options: Options passed to AerSimulator."""
+
+    run_options: dict = field(default_factory=dict)
+    """run_options: Options passed to run."""
+
+
+class EstimatorV2(BaseEstimatorV2):
+    """Evaluates expectation values for provided quantum circuit and observable combinations.
+
+    Run a fast simulation using Aer.
+    Sampling from a normal distribution ``N(expval, precison)`` when set to ``precision``.
+
+    * ``default_precision``: The default precision to use if none are specified in :meth:`~run`.
+      Default: 0.0.
+
+    * ``backend_options``: Options passed to AerSimulator.
+      Default: {}.
+
+    * ``run_options``: Options passed to :meth:`AerSimulator.run`.
+      Default: {}.
+    """
+
+    def __init__(
+        self,
+        *,
+        options: dict | None = None,
+    ):
+        """
+        Args:
+            options: The options to control the default precision (``default_precision``),
+                the backend options (``backend_options``), and
+                the runtime options (``run_options``).
+        """
+        self._options = Options(**options) if options else Options()
+        method = "density_matrix" if "noise_model" in self.options.backend_options else "automatic"
+        self._backend = AerSimulator(method=method, **self.options.backend_options)
+
+    @property
+    def options(self) -> Options:
+        """Return the options"""
+        return self._options
+
+    def run(
+        self, pubs: Iterable[EstimatorPubLike], *, precision: float | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        if precision is None:
+            precision = self._options.default_precision
+        coerced_pubs = [EstimatorPub.coerce(pub, precision) for pub in pubs]
+        self._validate_pubs(coerced_pubs)
+        job = PrimitiveJob(self._run, coerced_pubs)
+        job._submit()
+        return job
+
+    def _validate_pubs(self, pubs: list[EstimatorPub]):
+        for i, pub in enumerate(pubs):
+            if pub.precision < 0.0:
+                raise ValueError(
+                    f"The {i}-th pub has precision less than 0 ({pub.precision}). ",
+                    "But precision should be equal to or larger than 0.",
+                )
+
+    def _run(self, pubs: list[EstimatorPub]) -> PrimitiveResult[PubResult]:
+        return PrimitiveResult([self._run_pub(pub) for pub in pubs])
+
+    def _run_pub(self, pub: EstimatorPub) -> PubResult:
+        circuit = pub.circuit.copy()
```

Comment:
I think `pub.circuit` should be compiled to `AerCompiler` here to avoid copy. This will be an optimization in next releases.